From 64f81b7fad227097f5e135d25a5574cfecf4d50d Mon Sep 17 00:00:00 2001
From: Guilherme Amadio <amadio@cern.ch>
Date: Fri, 18 May 2018 11:30:58 +0200
Subject: [PATCH] Remove declarations of crc32 and adler32
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

These break compilation with the error below:

xrootd-4.8.3/tests/XrdSsiTests/XrdShMap.cc:
    In function ‘int DoA32(const char*)’:
xrootd-4.8.3/tests/XrdSsiTests/XrdShMap.cc:418:34:
    error: expected initializer before ‘OF’
ZEXTERN uLong ZEXPORT adler32 OF((uLong adler, const Bytef *buf, uInt len));
                              ^^
---
 src/XrdSsi/XrdSsiShMam.cc     | 2 +-
 tests/XrdSsiTests/XrdShMap.cc | 3 ---
 2 files changed, 1 insertion(+), 4 deletions(-)

diff --git a/src/XrdSsi/XrdSsiShMam.cc b/src/XrdSsi/XrdSsiShMam.cc
index 013dc855..a4803d71 100644
--- a/src/XrdSsi/XrdSsiShMam.cc
+++ b/src/XrdSsi/XrdSsiShMam.cc
@@ -882,7 +882,7 @@ bool XrdSsiShMam::GetItem(void *data, const char *key, int hash)
 /******************************************************************************/
   
 int XrdSsiShMam::HashVal(const char *key)
-{  ZEXTERN uLong ZEXPORT crc32 OF((uLong crc, const Bytef *buf, uInt len));
+{
    uLong crc;
    int hval, klen = strlen(key);
 
diff --git a/tests/XrdSsiTests/XrdShMap.cc b/tests/XrdSsiTests/XrdShMap.cc
index bb642ac3..860fa7ad 100644
--- a/tests/XrdSsiTests/XrdShMap.cc
+++ b/tests/XrdSsiTests/XrdShMap.cc
@@ -415,7 +415,6 @@ void Explain(const char *what)
   
 int DoA32(const char *buff)
 {
-   ZEXTERN uLong ZEXPORT adler32 OF((uLong adler, const Bytef *buf, uInt len));
    uLong adler = adler32(0L, Z_NULL, 0);
 
 // Check for ID request now
@@ -438,8 +437,6 @@ int DoA32(const char *buff)
   
 int DoC32(const char *buff)
 {
-   ZEXTERN uLong ZEXPORT crc32 OF((uLong crc, const Bytef *buf, uInt len));
-
 // Check for ID request now
 //
    if (!buff) {int myID; memcpy(&myID, "c32 ", sizeof(int)); return myID;}
-- 
2.17.0

